source/pool/contracts/Pool.sol

```diff
@@ -112,8 +112,8 @@ contract Pool is Ownable {
       claimed[claim.root][msg.sender] = CLAIMED;
     }
     uint256 amount = calculate(totalScore, token);
-    IERC20(token).transfer(msg.sender, amount);
     emit Withdraw(rootList, msg.sender, token, amount);
```

Comment:
I think events should be at the end, as it doesn't really affect security and makes it easier to reason about when getting all the events in a tx

Reply:
I agree. Seeing it as the last event in a trx makes things really easy. Will change.